radix-engine/tests/determinism.rs

```diff
@@ -17,3 +24,58 @@ fn test_simple_deterministic_execution() {
         .substate_store()
         .assert_eq(test_runner1.substate_store());
 }
+
+#[test]
+fn same_executions_result_in_same_final_state_hash() {
+    let state_hashes = (0..5)
+        .map(|_| create_and_pass_multiple_proofs())
+        .collect::<HashSet<Hash>>();
+    assert_eq!(
+        state_hashes.len(),
+        1,
+        "non-deterministic final state hash: {:?}",
+        state_hashes
+    );
+}
+
+/// Simulates a complete "test" which creates multiple proofs and passes them to a method.
```

Comment:
@talekhinezh plz see the impl and let me know if this is what you meant:
<img width="663" alt="image" src="https://user-images.githubusercontent.com/119413677/214541127-802bf018-6d3e-4642-b88a-941e3c1609e7.png">

Reply:
Does this test fail or succeed?

@talekhinezh  - how do we expect the non-determism to manifest?

I guess this non-determinism might only come from the receipt; so eg a (consensus) receipt hash accumulator might show us it? (but maybe not the state root hash?)

Reply:
it succeeds, 100% for me (not even flakey)
maybe the non-det bugs are not so easy to catch? only occur across machines?

anyways, I am treating this test just as an example for "how to use that new test util (for things that you suspect are non-det)".

We can also hope @talekhinezh has some pointers on how to make this test more flakey :) (I am not 100% sure that I implemented his idea correctly)